crates/hir-ty/src/infer/coerce.rs

```diff
@@ -673,7 +668,9 @@ impl InferenceTable<'_> {
                 // FIXME need to record an obligation here
                 canonicalized.apply_solution(self, subst)
             }
-            // FIXME actually we maybe should also accept unknown guidance here
+            Solution::Ambig(Guidance::Unknown) if must_be_unsized => {
+                // FIXME need to record an obligation here too
```

Comment:
Lets link https://github.com/rust-lang/rust-analyzer/issues/11847 here as a comment and state that this is a workaround

Reply:
Sure.

Reply:
Done.